src/rust/inetstack/protocols/udp/datagram/mod.rs

```diff
@@ -38,7 +38,7 @@ pub struct UdpDatagram {
     /// UDP header.
     udp_hdr: UdpHeader,
     /// Payload
-    data: DemiBuffer,
+    data: Option<DemiBuffer>,
```

Comment:
If the code panics when `data` is `None`, then why make it an `Option`?

Reply:
We discussed offline. The PacketBuf may be valid when `new` is called, but will be `None` after `take_body` is called. So these checks are necessary.